sdks/java/io/kafka/src/test/java/org/apache/beam/sdk/io/kafka/ReadFromKafkaDoFnTest.java

```diff
@@ -116,10 +118,14 @@ public void setCurrentPos(long pos) {
       this.currentPos = pos;
     }
 
-    public void setStartOffsetForTime(long pos) {
+    public void setStartOffsetForTime(KV<Long, Instant> pos) {
```

Comment:
Why not make this method take two parameters and construct its own internal representation using the KV as necessary.

Please do the same for `setStopOffsetForTime`